### PR TITLE
fix(go): clean up invalid extension uploads

### LINKED ISSUE #2725: Go SDK leaks an uploaded Browserbase extension when upload-response validation fails
﻿## Description

The Go Browserbase launch path uploads an owned Stagehand extension and then validates the full upload response. If validation fails while the response still contains a usable extension ID (for example, `createdAt` is missing or `fileName` is empty), `createSession()` returns immediately without deleting the extension that was just created.

Other failure paths after a successful upload call `deleteExtensionBestEffort`, but the response-validation path does not. Repeated transient/malformed API responses can therefore accumulate orphaned extensions on the Browserbase account.

## Code reference

- `packages/sdk-go/browserbase_session.go:85-97` uploads the extension and returns directly on `extension.validate()` failure.
- `packages/sdk-go/browserbase_session.go:106-115` correctly deletes the owned extension when session creation fails.
- `packages/sdk-go/browserbase_session.go:117-130` also cleans up after an invalid session response.
- `packages/sdk-go/browserbase_client.go:724-748` shows that validation can fail for fields other than `id`, leaving a valid ID available for cleanup.

## Reproduction

Add a focused test using `fakeBrowserbaseAPI` in `packages/sdk-go/browserbase_session_test.go`:

1. Make `uploadExtensionFunc` return a response with `ID: testPointer("ext_orphaned")` but one invalid/missing metadata field, such as `CreatedAt: nil`.
2. Call `client.createSession(context.Background(), BrowserbaseLaunchOptions{})`.
3. Assert that creation fails validation.
4. Inspect `api.deleteExtensionCalls`.

`deleteExtensionCalls` remains `0`, although the server-side extension ID is known.

## Expected behavior

When an upload succeeded and returned a non-empty extension ID, every subsequent local validation failure should make a best-effort deletion using a non-canceled cleanup context, while preserving the validation error (and joining any cleanup error).

## Actual behavior

The function returns the validation error and leaves the uploaded extension behind.

## Why it matters

Browserbase extensions are account resources. A malformed or partially incompatible API response should not turn a failed browser launch into a persistent resource leak, especially for applications that retry launch automatically.

## Duplicate/history check

I searched open and closed issues/PRs for Browserbase extension cleanup, invalid upload responses, and extension leaks. Existing tests cover upload errors, empty IDs, session-creation cleanup, and retrying failed deletion, but I found no issue or PR covering cleanup after upload-response validation fails.

### PR DESCRIPTION
**Pull request by abhinavkr26104**
# why

The Go SDK validates Browserbase's extension upload response after the upload has already created an account resource. If metadata validation failed while a usable extension ID was present, `createSession` returned without deleting the owned extension.

Fixes #2725

# what changed

- Capture a non-empty uploaded extension ID before validating the remaining response metadata.
- Best-effort delete that owned extension with a non-canceled context when validation fails.
- Join cleanup failures with the original validation error.
- Avoid issuing deletion requests when the response contains no usable extension ID.
- Add a Go SDK patch changeset.

# test plan

- `go test $(go list ./... | grep -v '/examples$')` (all Go SDK packages pass)
- `go -C packages/sdk-go/internal/generator test ./...`
- `go vet` for the Go SDK and generator packages
- `gofmt -l packages/sdk-go/browserbase_session.go packages/sdk-go/browserbase_session_test.go`
- `pnpm exec tsx scripts/release/check-changesets.ts`

The full monorepo suites were also attempted on Windows. Unrelated existing failures remain in cross-platform path assertions, docs link validation, generated extension archive comparison, and the checkout-wide CRLF formatting check.

---
## Summary by cubic
Deletes owned Browserbase extensions when upload response validation fails and clarifies cleanup errors. Previously, `createSession` could orphan an uploaded extension; now cleanup runs best-effort and reports the failure context.

- Capture a non-empty extension ID before validation and delete using a non-canceled context when validation fails.
- Skip deletion when there is no usable extension ID or the SDK does not own the extension.
- Join cleanup failures with the validation error and use contextual messages for validation vs. session failures.
- Add tests for invalid-response cleanup and avoiding deletes on empty IDs; add a patch changeset for `@browserbasehq/stagehand-go`.

<sup>Written for commit 5b85ccf6db57ba544966071695eaf4e329a83a82. Summary will update on new commits.</sup>

<a href="https://cubic.dev/pr/browserbase/stagehand/pull/2729?utm_source=github" target="_blank" rel="noopener noreferrer" data-no-image-dialog="true"><picture><source media="(prefers-color-scheme: dark)" srcset="https://www.cubic.dev/buttons/review-in-cubic-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://www.cubic.dev/buttons/review-in-cubic-light.svg">Review in cubic</picture></a>

### CODE REVIEW

File: packages/sdk-go/browserbase_session.go [hunk 1]
```diff
@@ -89,13 +89,19 @@ func (client *browserbaseSessionClient) createSession(
...
+		if extension.ID != nil {
+			extensionID = strings.TrimSpace(*extension.ID)
+		}
 		if err := extension.validate(); err != nil {
-			return resolvedBrowserSource{}, fmt.Errorf(
-				"validate Browserbase extension upload: %w",
-				err,
+			return resolvedBrowserSource{}, errors.Join(
+				fmt.Errorf("validate Browserbase extension upload: %w", err),
+				client.deleteExtensionBestEffort(
```
[the remark this answers is not in the document]
Comment (abhinavkr26104) [reply packages/sdk-go/browserbase_session.go:98]: Addressed in 5b85ccf6. The cleanup helper now accepts a path-specific sanitized message: upload-response validation failures report that upload validation failed, while session-creation failures retain the existing session-failure wording. The regression assertion covers the upload-specific message.
